docs/readmes/bazel/bazel.md

```diff
@@ -0,0 +1,510 @@
+
+Some useful command line options for unit testing include:
+
+- `--cache_test_results=no` disables the caching of test results and forces a re-run even if there are no changes.
+- `--runs_per_test=<integer>` executes all requested test targets exactly `<integer>`-many times, even if runs are successful. This flag can be used e.g. to detect flaky tests.
+- `--flaky_test_attempts=<integer>` executes all test targets that have the `flaky = True` attribute set, up to `<integer>`-many times. If there are failures the test is marked as flaky in the results. Test targets with the attribute `flaky = False` (default) are unaffected.
+
+A complete list of [command line arguments](https://bazel.build/reference/command-line-reference) can be found in the official bazel documentation.
+
+### LTE integration tests
+
+Information on how to run the LTE integration tests with bazel is documented on the [s1ap tests](../lte/s1ap_tests.md) page.
+
+> **Warning**: Do not execute bazel commands as sudo/root user, because it can break your local bazel setup!
+
+The LTE integration tests cannot be executed directly with bazel because the tests need extensive access to system resources that are not available during the bazel runtime. In CI the LTE integration tests are executed directly with pytest. Locally there is a wrapper script that can be used, for instructions on how to use this script please refer to the [s1ap tests](../lte/s1ap_tests.md) page.
+
+### Python sudo tests
+
+The Python sudo tests are a group of tests for the pipelined and mobilityd services that require root permissions during the test runtime.
+
+> **Warning**: Do not execute bazel commands as sudo/root user, because it can break your local bazel setup!
+
+To run all Python sudo tests, start the magma-dev VM and run the script:
+
+```bash
+cd $MAGMA_ROOT
+bazel/scripts/run_sudo_tests.sh
+```
+
+To run individual tests specify the test target, for example
+
+```bash
+cd $MAGMA_ROOT
+bazel/scripts/run_sudo_tests.sh //lte/gateway/python/magma/pipelined/tests:test_classifier_mme_flow_dl
+```
+
+To display the full list of supported options, run
+
+```bash
+cd $MAGMA_ROOT
+bazel/scripts/run_sudo_tests.sh --help
+```
+
+### Load tests
+
+> **Info**: For detailed information on the load tests please refer to the README file `lte/gateway/python/load_tests/README.md`.
+
+To run every load test that is defined inside the `LOAD_TEST_LIST` of the script `bazel/scripts/run_load_tests.sh`, run the script on the magma-dev VM from anywhere inside the Magma folder.
+
+For example, for mobilityd's `AllocateIPRequest`, there is a `load_test_mobilityd.py` which contains an `allocate` command (run as `load_test_mobilityd.py allocate`).
+
+The load tests are defined in bazel as `py_binary` targets, because they run as scripts and not as tests.
+
+### Generating unit test coverage
```

Comment:
Even if this is already mentioned in the bazel command dictionary, should we add something like ~"The data can be further processed with, e.g., genhtml via genhtml bazel-out/_coverage/_coverage_report.dat --output-directory path/to/output/directory." - this was a slack support query I got recently.

Reply:
Paragraph added